mitreattack/navlayers/core/README.md

```diff
@@ -0,0 +1,245 @@
+    _LayerObj().description                   # A description string for the Layer
+    _LayerObj().domain                        # The domain for the Layer
+    _LayerObj().filters                       # Link to a Filter object instance
+    _LayerObj().sorting                       # An integer denoting which sorting form to use
+    _LayerObj().layout                        # Link to a Layout object instance
+    _LayerObj().hideDisabled                  # Bool determining whether or not to show disabled techniques
+    _LayerObj().techniques                    # List of links to Technique objects
+    _LayerObj().gradient                      # Link to Gradient object
+    _LayerObj().legendItems                   # List of links to LegendItems objects
+    _LayerObj().showTacticRowBackground       # Bool determining whether or not to show a background for tactics
+    _LayerObj().tacticRowBackground           # Color code for tactic background
+    _LayerObj().selectTechniquesAcrossTactics # Bool determining whether or not to select cross-tactic
+    _LayerObj().selectSubtechniquesWithParent # Bool determining whether or not to select subtechniques
+    _LayerObj().metadata                      # List of links to Metadata items
+    _LayerObj().get_dict()                    # Export Layer as a dictionary object
+```
+#### Versions Object
+```python
+    Versions().layer      # String denoting Layer format version
+    Versions().__attack   # String denoting ATT&CK version
+    Versions().navigator  # String denoting Navigator version
+    Versions().get_dict() # Export Version data as a dictionary object
+```
+#### Filter Object
+```python
+    Filter().domain     # String denoting the domain for the Filter
+    Filter().platforms  # String denoting platforms within this filter
+    Filter().get_dict() # Export Filter data as a dictionary object
+```
+Please note that although not documented here, there is another Filter object variant, Filterv3, which exists
+for backwards compatibility reasons.
+#### Layout Object
+```python
+    Layout().layout              # String denoting which layout form to use
+    Layout().showID              # Bool denoting whether or not to show technique IDs
+    Layout().showName            # Bool denoting whether or not to show technique names
+    Layout().showAggregateScores # Bool denoting whether or not to utilize Aggregate scores
+    Layout().countUnscored       # Bool denoting whether ot not to count unscored techniques as 0s for Aggregates
+    Layout().aggregateFunction   # A enum integer denoting which aggregate function to utilize
+                                 # 1 - Average, 2 - min, 3 - max, 4 - sum
+    Layout().get_dict()          # Export Layout data as a dictionary object
+    Layout().compute_aggregate() # Compute the aggregate score for a technique and it's subtechniques
+    Layout().aggFunction()       # Wrapper handle to apply the selected aggregate function to data
```

Comment:
The documentation here does not properly distinguish the use of these functions which is only apparent when reading the actual function docstring. `aggFunction` may be better as a private (e.g starting with underscore and not listed here) function since it's really just a helper for `compute_aggregate`.

Reply:
That definitely makes sense. Originally I think I left it this way because of some attempts to be clever with calling that function, but looking back over the code there's no reason to not make it 'private'.